server/multitenancy/tenant_resolver.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { KibanaRequest } from '../../../../src/core/server';
+import { SecurityPluginConfigType } from '..';
+import { SecuritySessionCookie } from '../session/security_cookie';
+import { isEmpty, findKey, cloneDeep } from 'lodash';
+
+/**
+ * Resovles the tenant the user is using.
+ *
+ * @param request Kibana request.
+ * @param config security plugin config.
+ * @param cookie cookie extracted from the request. The cookie should have been parsed by AuthenticationHandler.
+ * pass it as parameter instead of extracting again.
+ * @param authInfo authentication info, the Elasticsearch authinfo API response.
+ *
+ * @returns user preferred tenant of the request.
+ */
+
+export const PRIVATE_TENANT: string = '__user__';
+export const GLOBAL_TENANT: string = '';
+
+export function resolveTenant(
+  request: KibanaRequest,
+  username: string,
+  availabeTenants: any,
+  config: SecurityPluginConfigType,
+  cookie: SecuritySessionCookie
+): string {
+  let selectedTenant: string = undefined;
+  if (request.query && ((request.query as any).security_tenant || (request.query as any).securitytenant)) {
```

Comment:
request.query as any has been used many time
could we define ```const query: any = request.query to improve readability

Reply:
updated accordingly